src/Sylius/Behat/Service/Provider/RequestApiPathPrefixProvider.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Behat\Service\Provider;
+
+use Sylius\Bundle\ApiBundle\Context\UserContextInterface;
+use Sylius\Bundle\ApiBundle\Provider\ApiPathPrefixProvider;
+use Sylius\Bundle\ApiBundle\Provider\ApiPathPrefixProviderInterface;
+use Sylius\Bundle\ApiBundle\Provider\RequestApiPathPrefixProviderInterface;
+use Sylius\Component\Core\Model\AdminUserInterface;
+use Sylius\Component\Core\Model\ShopUserInterface;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\Security\Core\User\UserInterface;
+
+/** @experimental */
+final class RequestApiPathPrefixProvider implements RequestApiPathPrefixProviderInterface
```

Comment:
I added this class because in a few places on behat contexts we are using IriConverter without request and it this raises problems, but this service should fix it.

Reply:
I added new approach to test it look in step: https://github.com/Sylius/Sylius/pull/11925/files#diff-3ea3a89418303ef08443a4292dd9cd26b55365f535efadb9762e47d82631d106R21